### DISCUSSION TITLE
Trouble with hmApp.alarmNew

### DISCUSSION THREAD

**Discussion by HeroCod** [zepp-health/zeppos-forum#355] [Q&A] [answered]
Hello, it's my first time posting here so i hope the topic and format is correct.

I'm currently trying to get a miniProgram to work on my systems, with zeppOS 3.5, which is supposed to execute a task to refresh some sensors on an external platform with data from the device every 10 minutes.

Currently the way i'm trying to implement this application, has led me to have some problems with the hmApp.alarmNew() method and its (lack of) descriptive documentation on its limitations and parameters it accepts: 

with something like

```
const alarm = hmApp.alarmNew({
        file: "page/sensors_update/index.page",
        appid: appId,
        delay: 600,
        param: FS_REF_SENSORS_UPDATE_ALARM_ID,
      });
```

my smartwatch does not seem to be able to open back the app for the sensor update task on its own, is there a time limit on how often an app can self-open? Does any documentation of this method, it's parameters and functionality, exist?

I tried looking in slack, GitHub, the official docs, etc, but nothing seems to come up, and definitely not any kind of per-function documentation, just examples sprinkled around.

Sorry for the frustrated tone, but i really could get my hands on any proper documentation so far.

**Comment by SashaCX75** [comment, marked as answer]
For ZeppOS 3.5 it's probably better to use setLaunchAppTimeout.
https://docs.zepp.com/docs/reference/device-app-api/newAPI/router/setLaunchAppTimeout/
On the same site you can find documentation on the functions.

I also recommend to read about "App Service" and @zos/alarm
https://docs.zepp.com/docs/guides/framework/device/app-service/
